config/llmisvc/config-llm-decode-worker-data-parallel.yaml

```diff
@@ -69,10 +69,50 @@ spec:
           - containerPort: 8001
             protocol: TCP
         command:
-          - "/bin/sh"
+          - "/bin/bash"
           - "-c"
         args:
           - |-
+            
+            if [ "$KSERVE_INFER_ROCE" = "true" ]; then
+              echo "Trying to infer RoCE configs ... "
+              grep -H . /sys/class/infiniband/*/ports/*/gid_attrs/types/* 2>/dev/null
+            
+              KSERVE_INFER_IB_GID_INDEX_GREP=${KSERVE_INFER_IB_GID_INDEX_GREP:-"RoCE v2"}
+              KSERVE_INFER_IB_GID_INDEX_INFINIBAND=${KSERVE_INFER_IB_GID_INDEX_INFINIBAND:-${NCCL_IB_HCA}}
+
+              GID_DIR="/sys/class/infiniband/${KSERVE_INFER_IB_GID_INDEX_INFINIBAND}/ports/1"
+              GID_INDEX=""
+            
+              # Prefer IPv4
+              for tpath in ${GID_DIR}/gid_attrs/types/*; do
+                idx=$(basename "$tpath")
+                typ=$(cat "$tpath" 2>/dev/null || true)
+                if [ "$typ" = "${KSERVE_INFER_IB_GID_INDEX_GREP}" ]; then
+                  gid=$(cat ${GID_DIR}/gids/$idx 2>/dev/null || true)
+                  echo "[Infer RoCE] Evaluating ${tpath} with IP=${gid}"
+                  if echo "$gid" | grep -q ':ffff:'; then
+                    GID_INDEX="$idx"; break
+                  fi
+                fi
+              done
+            
+              # Pick the first one if IPv4 is not found
+              if [ -z "$GID_INDEX" ]; then
+                echo "[Infer RoCE] No IPv4 GID_INDEX found for ${GID_DIR}, using the first one available ... "
+                GID_INDEX=$(grep -H . ${GID_DIR}/gid_attrs/types/* 2>/dev/null | grep "${KSERVE_INFER_IB_GID_INDEX_GREP}" | awk -F'[/:]' '{print $10; exit}')
+              fi
+
+              if [ -n "$GID_INDEX" ]; then
+                echo "[Infer RoCE] Setting inferred *_IB_GID_INDEX=${GID_INDEX}"
+                export NCCL_IB_GID_INDEX=${NCCL_IB_GID_INDEX:-$GID_INDEX}
+                export NVSHMEM_IB_GID_INDEX=${NVSHMEM_IB_GID_INDEX:-$GID_INDEX}
+                export UCX_IB_GID_INDEX=${UCX_IB_GID_INDEX:-$GID_INDEX}
+              else
+                echo "[Infer RoCE] Skipping RoCE infer export, no GID_INDEX found."
+              fi
+            fi
```

Comment:
Do we want to make this explicit and duplicated here for easier tweaking by the end user?
Alternatively, we could embed this script with `//go:embed` and include it in the global config struct used for templating.
WDYT?

Reply:
Let's track this as an improvement to follow up?